src/libraries/System.Private.CoreLib/src/System/Text/DecoderFallback.cs

```diff
@@ -11,15 +11,12 @@ namespace System.Text
 {
     public abstract class DecoderFallback
     {
-        private static DecoderFallback? s_replacementFallback; // Default fallback, uses no best fit & "?"
-        private static DecoderFallback? s_exceptionFallback;
+        // Default fallback, uses no best fit & "?"
+        private static readonly DecoderReplacementFallback s_replacementFallback = new DecoderReplacementFallback();
+        public static DecoderFallback ReplacementFallback => s_replacementFallback;
```

Comment:
```suggestion
        public static DecoderFallback ReplacementFallback { get; } = new DecoderReplacementFallback();
```

Reply:
I noticed that using this pattern the private field generated by the compiler is typed as `DecoderFallback`, not `DecoderReplacementFallback`. Would this impact devirtualization?

Reply:
The way to do this in IL linker way is to define the singleton inside the class that the singleton belongs to. Ie, this would become: `public static DecoderFallback ReplacementFallback => DecoderReplacementFallback.s_default;`.

Reply:
This is passed through so many layers that the devirtualization does not matter.

But you can still keep the precise type of the field if you implement the other suggestion.

Reply:
> I noticed that using this pattern the private field generated by the compiler is typed as DecoderFallback, not DecoderReplacementFallback

FYI, https://github.com/dotnet/roslyn/issues/30797